src/persist-client/src/lib.rs

```diff
@@ -54,6 +54,26 @@ pub(crate) mod r#impl {
 // TODOs
 // - Decide if the inner and outer error of the two-level errors should be
 //   swapped.
+// - Split errors into definitely failed vs maybe failed variants. More
```

Comment:
we probably want todos for physical/logical compaction and then garbage collection?

Reply:
good call! done